pkg/collect/host_network.go

```diff
@@ -20,13 +23,67 @@ const (
 	NetworkStatusConnected            = "connected"
 	NetworkStatusErrorOther           = "error"
 	NetworkStatusBindPermissionDenied = "bind-permission-denied"
+	NetworkStatusInvalidAddress       = "invalid-address"
 )
 
 type NetworkStatusResult struct {
-	Status NetworkStatus `json:"status"`
+	Status  NetworkStatus `json:"status"`
+	Message string        `json:"message"`
+}
+
+func isValidIPCandidate(address string) bool {
```

Comment:
Why not use a regex?

Reply:
Mostly because of an unfamiliarity with regex. Not sure if we prefer regex vs code/algorithms.